boa_engine/src/builtins/function/mod.rs

```diff
@@ -192,7 +196,7 @@ impl Function {
     pub fn is_constructor(&self) -> bool {
         match self {
             Self::Native { constructor, .. } | Self::Closure { constructor, .. } => *constructor,
-            Self::VmOrdinary { code, .. } => code.constructor,
+            Self::VmOrdinary { code, .. } | Self::VmGenerator { code, .. } => code.constructor,
```

Comment:
Maybe we should rename `VmOrdinary` => `Ordinary` and `VmGenerator` => `Generator`, This is an artifact from when we had an AST walker and I named it with a `Vm` prefix so we could have the vm and AST walker and work on both at the same time. But now I don't see a reason to keep that naming since there is only a vm now.